crates/aptos/src/config/mod.rs

```diff
@@ -179,6 +179,23 @@ fn global_folder() -> CliTypedResult<PathBuf> {
     }
 }
 
+fn find_workspace_config(starting_path: PathBuf) -> CliTypedResult<PathBuf> {
+    let mut current_path = starting_path.clone();
+    loop {
+        let cand = current_path.join(CONFIG_FOLDER);
+        if cand.is_dir() {
```

Comment:
Note this is going to make a slightly strange behavior.  Have you tried seeing what happens if you have nothing above you and you run `aptos init`?  I think it should work, but I wasn't sure if it would keep updating it no matter what.

Reply:
Good point. This let me rewrite the logic a bit, we really need to distinguish whether we want this search behavior or not, depending on the command.